### PR TITLE
New parsers

### LINKED ISSUE #93: Add subparsers to `train slice`
`train slice` is not homogeneous with others because it does not require a network type. `cnn` subparser could be added, as well as `multicnn` as a new feature.

### PR DESCRIPTION
**Pull request by 14thibea**
Add `slice` subparsers (autoencoder / cnn / multicnn) to have a better homogeneity with other parsers.
`roi` was also enriched with multicnn.

### CODE REVIEW

File: docs/Train/ROI.md [hunk 1]
```diff
@@ -120,7 +124,7 @@ This argument will only be taken into account if the source network is a CNN.
 Choices are `best_loss` and `best_balanced_accuracy`. Default: `best_balanced_accuracy`.
 - `--selection_threshold` (float) threshold on the balanced accuracies to compute the 
 [image-level performance](./Details.md#soft-voting). 
-Patches are selected if their balanced accuracy is greather than the threshold. Default corresponds to no selection.
+Regions are selected if their balanced accuracy is greather than the threshold. Default corresponds to no selection.
```
Comment (mdiazmel) [inline docs/Train/ROI.md]:
```suggestion
Regions are selected if their balanced accuracy is greater than the threshold. Default corresponds to no selection.
```

File: docs/Train/ROI.md [hunk 2]
```diff
@@ -160,3 +164,100 @@ results
     Region-level performance corresponds to an evaluation in which both ROI are considered to be independent.
     However it is not the case, and what is more interesting is the evaluation at the image-level, 
     for which the predictions of the two regions were [assembled](./Details.md#soft-voting).
+
+
+## `train roi multicnn` - Train one classification CNN per region
+
+Contrary to the preceding network in which all regions of interest were used as input of a unique CNN, with this option
```
Comment (mdiazmel) [inline docs/Train/ROI.md]:
```suggestion
Contrary to the preceding network in which all regions of interest were used as input of a unique CNN, this option
```

File: docs/Train/ROI.md [hunk 3]
```diff
@@ -160,3 +164,100 @@ results
...
+a CNN is trained per region. Then the predictions of the CNNs are [assembled](./Details.md#soft-voting) to determine
```
Comment (mdiazmel) [inline docs/Train/ROI.md]: This suggestion follows the previous one. 

```suggestion
allows to train a CNN per region. Then the predictions of the CNNs are [assembled](./Details.md#soft-voting) to determine
```

File: docs/Train/ROI.md [hunk 4]
```diff
@@ -160,3 +164,100 @@ results
...
+the label at the image level.
+
+The output of each CNN is a vector of size equals to the number of classes in this dataset.
+This vector can be preprocessed by the [softmax function](https://pytorch.org/docs/master/generated/torch.nn.Softmax.html) 
+to produce a probability for each class. During training, the CNN is optimized according to the cross-entropy loss, 
+which becomes null for a subset of images if the CNN outputs 100% probability for the true class of each image of the subset.
```
Comment (mdiazmel) [inline docs/Train/ROI.md]: I will change this text, to make it more clear:

> "During the training, the CNN is optimized according to the cross-entropy loss. This value becomes zero (or null?) for a subset of images if the probability of the CNN is 1, with respect to the true class (ground truth) of each image in the subset."

File: docs/Train/Slice.md [hunk 5]
```diff
@@ -11,7 +11,86 @@ of a ResNet-18 trained on ImageNet.
     It is possible to add a custom architecture and train it with `clinicadl`.
     Detailed instructions can be found [here](./Custom.md).
 
-## Introduction
+## `train slice autoencoder` - Train autoencoders using 2D slices
+
+The objective of an autoencoder is to learn to reconstruct images given in input while performing a dimension reduction. 
+Slices at the beginning or at the end of the volume may be excluded using the `discarded_slices` argument.
+
+The difference between the input and the output image is given by the mean squared error.
```
Comment (mdiazmel) [inline docs/Train/Slice.md]:
```suggestion
The difference between the input and the output image is given by the mean squared error (MSE).
```

File: docs/Train/Slice.md [hunk 6]
```diff
@@ -22,11 +101,99 @@ This vector can be preprocessed by the [softmax function](https://pytorch.org/do
 to produce a probability for each class. During training, the CNN is optimized according to the cross-entropy loss,
```
Comment (mdiazmel) [inline docs/Train/Slice.md]: I know that you didn't introduce this phrase in this PR, but let's profit of this PR to reformulate this text as above (see  ` docs/Train/ROI.md`), if you're agree.

**Review by mdiazmel — COMMENTED** [review]
Hi @14thibea ,
Thanks for the PR!

I added some text proofing and suggestions to reformulate some texts. 

I thing we can merge once these items are solved.

File: docs/Train/Slice.md (hunk 6 above)
Comment (14thibea) [reply docs/Train/Slice.md]: Done in https://github.com/aramis-lab/AD-DL/pull/119/commits/ca0ae2054e0ae5f73ab4de3084096140af19f0f9

File: docs/Train/ROI.md (hunk 4 above)
Comment (14thibea) [reply docs/Train/ROI.md]: Done in https://github.com/aramis-lab/AD-DL/pull/119/commits/ca0ae2054e0ae5f73ab4de3084096140af19f0f9

**Review by mdiazmel — COMMENTED** [review]
I'll merge this PR.
